client/src/components/Upload.vue

```diff
@@ -64,50 +103,72 @@ export default {
       }
       var uploaded = [];
       this.$emit("update:uploading", true);
-      await Promise.all(
-        this.pendingUploads.map(async pendingUpload => {
-          var { name, files, fps } = pendingUpload;
-          fps = parseInt(fps);
-          pendingUpload.uploading = true;
-          var { data: folder } = await this.girderRest.post(
-            "/folder",
-            `metadata=${JSON.stringify({
-              viame: true,
-              fps,
-              type: pendingUpload.type
-            })}`,
-            {
-              params: {
-                parentId: this.location._id,
-                name
-              }
-            }
-          );
-          var pending = files;
-          var results = [];
-          while (pending.length) {
-            results = results.concat(
-              await Promise.all(
-                pending.splice(0, 500).map(async file => {
-                  var uploader = new Upload(file, {
-                    $rest: this.girderRest,
-                    parent: folder
-                  });
-                  return await uploader.start();
-                })
-              )
-            );
-          }
-          uploaded.push({ folder, results, pipeline: pendingUpload.pipeline });
-          this.remove(pendingUpload);
-        })
-      );
+
+      // This is in a while loop to act like a Queue with it adding new items during upload
+      while (this.pendingUploads.length > 0) {
+        await this.uploadPending(this.pendingUploads[0], uploaded);
+      }
       this.$emit("update:uploading", false);
       this.$emit("uploaded", uploaded);
+    },
+    async uploadPending(pendingUpload, uploaded) {
+      var { name, files, fps } = pendingUpload;
+      fps = parseInt(fps);
+      pendingUpload.uploading = true;
+      var { data: folder } = await this.girderRest
+        .post(
+          "/folder",
+          `metadata=${JSON.stringify({
+            viame: true,
+            fps,
+            type: pendingUpload.type
+          })}`,
+          {
+            params: {
+              parentId: this.location._id,
+              name
+            }
+          }
+        )
+        .catch(error => {
+          if (
+            error.response &&
+            error.response.data &&
+            error.response.data.message
+          ) {
+            this.errorMessage = error.response.data.message;
+          } else {
+            this.errorMessage = error;
+          }
+          pendingUpload.uploading = false;
+          // Return empty object for the folder destructuring
+          return { data: null };
+        });
```

Comment:
Since you're already awaitng the girderRest call, can't this can be handled with a `try-catch` around the await? This would make it easier to read as well as more standard for using await.

Reply:
I can do that, but I think it prevents the assignment destructuring of the returned value from the await.
`        let { data: folder } = await this.girderRest.post(......`
 The options I can think of are either to wrap the entire data and where it used inside the try statement (including other awaits), hoist the variable declaration with a 'var' (don't want to do) or declare the variable outside of the try block and then grab the data afterwards using assignment without declaration syntax (essentially wrapping the entire await in ())
`let folder;
      try {
        ({ data: folder } = await this.girderRest.post(.......)`
let me know if you have another method that I might not be thinking of.

Reply:
Not messing with the code formatted for github, here is what I'm talking about
![image](https://user-images.githubusercontent.com/61746913/77794332-1161dc80-7042-11ea-83c2-e5e48f52daec.png)

Reply:
In response to that code snippet, I think that'd be fine, or at least better than the alternative. If you didn't want to use the weird destructuring assignment syntax, you could also do this:

```
let folder;
try {
  const { data } = await this.girderRest.post(
  ...
  );
  folder = data;
}
```

And in the catch block you'd probably do something like
```
catch {
  ...
  folder = null;
}
```